src/hatti/ona/stream.cljs

```diff
@@ -0,0 +1,52 @@
+(ns hatti.ona.stream
+  (:require-macros [cljs.core.async.macros :refer [go]])
+  (:require [cljs.core.async :refer [<! put! chan timeout close!]]
+            [cljsjs.oboe]))
+
+(defn process-aggregated-data!
+  "Calls callback on aggregated data that come via messages in channel.
+   Callback accepts two arguments: data (cljs list), and completed? (bool)
+  If there is a parser in message, resume it after a timeout."
+  [agg-data-channel callback]
+  (go
+   (while true
+     (let [{:keys [data completed? ]} (<! agg-data-channel)]
+       (when data (callback data completed?))
+       ;; Timeout so browser can do some rendering before parsing again.
+       (<! (timeout 10))
+       (when completed?
+         (<! (timeout 1000))
+         (close! agg-data-channel))))))
+
+(defn read-next-chunk!
+  "On each step of data, increment a counter *n*, and put data in a list *agg*.
+   After a exponentially increasing # of steps have been processes,
+   the aggregated data gets flushed into a channel, and parser is paused.
+   The channel reader should resume the parser after processing the data."
+  [data-chunk *n* *agg* channel]
```

Comment:
`atom`s don't need earmuffs, https://github.com/bbatsov/clojure-style-guide#atoms, https://github.com/bbatsov/clojure-style-guide#earmuffs-for-dynamic-vars